x-pack/plugins/ml/public/application/jobs/new_job/pages/components/job_details_step/components/job_id/job_id_input.tsx

```diff
@@ -23,9 +23,18 @@ export const JobIdInput: FC = () => {
 
   useEffect(() => {
     const isEmptyId = jobId === '';
+    let valid = isEmptyId === true || jobValidator.jobId.valid;
+    let message = isEmptyId === false ? jobValidator.jobId.message : '';
+
+    valid = valid && jobValidator.latestValidationResult.jobIdExists?.valid === true;
+    message =
+      message === '' || message === undefined
+        ? jobValidator.latestValidationResult.jobIdExists?.message
+        : message;
+
```

Comment:
IMO splitting into multiple assignments doesn't make it easier to read

Reply:
also, the effect can be replaced with `useMemo` similar to `groups_input`

Reply:
> IMO splitting into multiple assignments doesn't make it easier to read
there are three outcomes, so without using a nested ternary, I can't see cleaner way to write it.

Reply:
ignore this ^
I believe it doesn't need the empty string outcome, `undefined` is fine for validation in the EUI component when it is valid.

Reply:
the empty string option isn't needed as the EUI component's validation will not show the message if the input is valid.